.setup/bin/setup_sample_courses.py

```diff
@@ -753,45 +753,59 @@ def create(self):
                             (gradeable.submission_due_date < NOW or random.random() < 0.5) and \
                             (random.random() < 0.9) and \
                             (max_submissions is None or submission_count < max_submissions)):
-
-                        os.system("mkdir -p " + os.path.join(submission_path, "1"))
-                        submitted = True
-                        submission_count += 1
-                        current_time_string = dateutils.write_submitty_date(gradeable.submission_due_date - timedelta(days=1))
-
-                        conn.execute(electronic_gradeable_data.insert(), g_id=gradeable.id, user_id=user.id,
-                                     g_version=1, submission_time=current_time_string)
-                        conn.execute(electronic_gradeable_version.insert(), g_id=gradeable.id, user_id=user.id,
-                                     active_version=1)
-                        with open(os.path.join(submission_path, "user_assignment_settings.json"), "w") as open_file:
-                            json.dump({"active_version": 1, "history": [{"version": 1, "time": current_time_string}]},
-                                      open_file)
-                        with open(os.path.join(submission_path, "1", ".submit.timestamp"), "w") as open_file:
-                            open_file.write(current_time_string + "\n")
-
-                        if isinstance(gradeable.submissions, dict):
-                            for key in gradeable.submissions:
-                                os.system("mkdir -p " + os.path.join(submission_path, "1", key))
-                                submission = random.choice(gradeable.submissions[key])
-                                src = os.path.join(gradeable.sample_path, submission)
-                                dst = os.path.join(submission_path, "1", key)
-                                create_gradeable_submission(src, dst)
-                        else:
-                            submission = random.choice(gradeable.submissions)
-                            if isinstance(submission, list):
-                                submissions = submission
+                        active_version = random.choice(range(1, versions_to_submit+1))
+                        json_history = {"active_version": active_version, "history": []}
+                        random_days = 1
+                        if random.random() < 0.3:
+                            random_days = random.choice(range(-3,2))
+                        for version in range(1, versions_to_submit+1):
+                            os.system("mkdir -p " + os.path.join(submission_path, str(version)))
+                            submitted = True
+                            submission_count += 1
+                            current_time_string = dateutils.write_submitty_date(gradeable.submission_due_date - timedelta(days=random_days))
```

Comment:
```bash
root@vagrant:/var/local/submitty/courses/s18/sample/submissions/open_homework/student# cat user_assignment_settings.json 
{
    "active_version": 2,
    "history": [
        {
            "version": 1,
            "type": "upload",
            "who": "student",
            "time": "9996-12-30 23:59:59-0500"
        },
        {
            "version": 2,
            "time": "2018-05-31 14:16:22-0400 America/New_York",
            "who": "student",
            "type": "upload"
        }
    ]
}root@vagrant:/var/local/submitty/courses/s18/sample/submissions/open_homework/student# cat 2/.submit.timestamp 
2018-05-31 14:16:22-0400 America/New_York
```
Timezone is missing. Not sure if it will be an issue.

Reply:
I noticed that before, but I couldn't figure out how to add the time zone. However I haven't encountered any issues without the timezone.

Reply:
The important part of the timezone is the -0500 and -0400.  The location is just for human-readability, its ok that its not there.  I am worried why they don't match. Are we hardcoding the 0500 somewhere?  (I think that's standard time and the 0400 is daylight savings)

Reply:
The timezone should probably be investigated, but I've approved the other changes since the courses are populated and seem consistent, as well as having a good variety of situations.